new_lineage_panel/src/App.tsx

```diff
@@ -92,9 +92,13 @@ export const LineageContext = createContext<{
   setMinRange: Dispatch<SetStateAction<[number, number]>>;
   nodeCount: number;
   setNodeCount: Dispatch<SetStateAction<number>>;
+  selectCheck: boolean;
```

Comment:
Food for thought: you can create a slice like this https://github.com/AltimateAI/vscode-dbt-power-user/blob/master/webview_panels/src/modules/app/appSlice.ts#L8 and use it in your context/provider 
https://github.com/AltimateAI/vscode-dbt-power-user/blob/master/webview_panels/src/modules/app/AppProvider.tsx#L17
Then you can dispatch any actions in slice and get state values easily in any component https://github.com/AltimateAI/vscode-dbt-power-user/blob/master/webview_panels/src/modules/dataPilot/components/queryAnalysis/provider/useQueryAnalysisContext.ts#L18. This will help to avoid setting all these getters and setters in context everytime